### ISSUE TITLE
Need last revision date

### ISSUE THREAD

**Issue by tkharris** [jmthrower/google-cells#3]
I'd like to get the last revision date now as well as the last revision user name. Maybe it is time to start thinking about pulling revisions and users into classes.

**Comment by jmthrower** [comment]
Cool. I can probably do some work on it this week.

**Comment by tkharris** [comment]
That would be great! Let me know if you start working on that.

**Comment by jmthrower** [comment]
@tkharris I have most of this functionality finished in this PR: https://github.com/jmthrower/google-cells/pull/5

Since there hasn't been a release since we merged in your other PR, what I'll do is remove Spreadsheet#get_last_modifying_user and release the new stuff instead in version 0.4.0.

To upgrade:

``` ruby
spreadsheet = GoogleCells::Spreadsheet.get('key')
revisions = spreadsheet.revisions
r = revisions.last
r.author.name
# => "Jane Smith"
r.author.email
# => "jane@example.com"
r.updated_at
# => "2014-09-17T16:38:52.598Z"
```

I maintained the default sort as chronological, so the last revision should still be the most recent. However, the logic to exclude service account emails and nil names will need to be reproduced on your end, as well as adding in "_unknown_" when no revisions exist. It didn't make sense to keep that for general usage, since some cases may need that info.

Something like this:

``` ruby
revisions = spreadsheet.revisions.reject{ |r| r.author.nil? }
revisions = revisions.reject{ |r| r.author.email =~ /gserviceaccount.com$/ }
r = revisions.last
return "*unknown*" if r.nil? || r.author.name.nil?
return r.author.name
# => "Jane Smith"
```

I'll also warn that the field "email" is not always present. Google seems to only provide that information when the account making the request is also the author, and otherwise leaves it blank. For your service account then, the logic to exclude its own revisions should work predictably, but otherwise I wouldn't rely on it.

**Comment by jmthrower** [comment]
Just so you know, I'll merge the PR and release a new version either tonight or tomorrow.

**Comment by tkharris** [comment]
Very cool! Those all sounds like sensible design decisions.

**Comment by jmthrower** [comment]
PR is merged; 0.4.0 is live on rubygems. Closing.
